contracts/voting-verifier/src/events.rs

```diff
@@ -145,55 +170,20 @@ pub struct TxEventConfirmation {
     pub payload_hash: [u8; 32],
 }
 
-impl TryFrom<Message> for TxEventConfirmation {
-    type Error = ContractError;
-
-    fn try_from(other: Message) -> Result<Self, Self::Error> {
-        let (tx_id, event_index) = parse_message_id(&other.cc_id.id)?;
-
-        Ok(TxEventConfirmation {
-            tx_id,
-            event_index,
-            destination_address: other.destination_address,
-            destination_chain: other.destination_chain,
-            source_address: other.source_address,
-            payload_hash: other.payload_hash,
-        })
-    }
-}
-
-fn parse_message_id(
-    message_id: &nonempty::String,
-) -> Result<(nonempty::String, u32), ContractError> {
-    // expected format: <tx_id>:<index>
-    let components = message_id
-        .split(TX_HASH_EVENT_INDEX_SEPARATOR)
-        .collect::<Vec<_>>();
-
-    if components.len() != 2 {
-        return Err(ContractError::InvalidMessageID(message_id.to_string()));
-    }
-
-    let event_index = components[1];
-    if event_index != "0" && event_index.starts_with('0') {
-        return Err(ContractError::InvalidMessageID(message_id.to_string()));
-    }
-
-    Ok((
-        components[0].try_into()?,
-        event_index
-            .parse()
-            .map_err(|_| ContractError::InvalidMessageID(message_id.to_string()))?,
-    ))
-}
-
-pub fn construct_message_id(tx_hash: Hash, event_index: u32) -> String {
-    format!(
-        "{}{}{}",
-        HexBinary::from(tx_hash),
-        TX_HASH_EVENT_INDEX_SEPARATOR,
-        event_index
-    )
+pub fn make_tx_event_confirmation(
```

Comment:
Consider `impl TryFrom<(Message, &MessageIdFormat)> for TxEventConfirmation`

Reply:
I thought of that, it just felt a bit messy. But if you think that's cleaner, I can reconsider.